### ISSUE TITLE
ochartsng depends on uninstallable packages

### ISSUE THREAD

**Issue by quirsh** [wellenvogel/ochartsng#12]
Hi,
I'm trying to install the ochartsng plugin as directed by the website using ```apt```.
However, the installation is stopped by apt because it claims that there are unmet dependencies: 
```
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 avnav-ochartsng : Depends: libssl1.1 but it is not installable
                   Depends: python3-mapbox-earcut but it is not installable or
                            base-files (< 12) but 12ubuntu4.6 is to be installed
E: Unable to correct problems, you have held broken packages.

```

My system is running Ubuntu 22.04 on a RaspberryPi4, I previously updated and upgraded all packages.

To my understanding libssl1.1 is not installable because it is deprecated and you can only install the new version libssl3.

Any help would be very much appreciated!

**Comment by free-x** [comment]
#11

**Comment by quirsh** [comment]
@free-x ah cool, many thanks. I checked the issues but not the PRs.
Is there a time frame when the merge is going to happen?

**Comment by free-x** [comment]
"python3-mapbox-earcut" needs to be backported also :-(

**Comment by quirsh** [comment]
What about ```base-files```? It is the alternative to ```python3-mapbox-earcut``` and at least on my system it is already installed:
```
base-files is already the newest version (12ubuntu4.6).
```

**Comment by free-x** [comment]
base-files Version for Ubuntu/jammy and Debian/bookworm are the same (12). But jammy ist forked from bookworm 1 year earlier.

**Comment by wellenvogel** [comment]
daily is building right now - should take ~2h

**Comment by free-x** [comment]
@quirsh 
how are the repositories of free-x.de integrated?
Please output from
```
grep -r free-x.de /etc/apt
```

**Comment by quirsh** [comment]
```
user@host:~$ grep -r free-x.de /etc/apt
/etc/apt/sources.list.d/extra.list:deb [signed-by=/usr/share/keyrings/oss.boating.gpg] https://www.free-x.de/debpreview bullseye main contrib non-free
/etc/apt/sources.list.d/extra.list:deb [signed-by=/usr/share/keyrings/oss.boating.gpg] https://www.free-x.de/debian bullseye main contrib non-free
/etc/apt/sources.list.d/extra.list:deb [signed-by=/usr/share/keyrings/oss.boating.gpg] https://www.free-x.de/ubuntu jammy main
/etc/apt/sources.list.d/extra.list.save:deb [signed-by=/usr/share/keyrings/oss.boating.gpg] https://www.free-x.de/debpreview bullseye main contrib non-free
/etc/apt/sources.list.d/extra.list.save:deb [signed-by=/usr/share/keyrings/oss.boating.gpg] https://www.free-x.de/debian bullseye main contrib non-free
```

**Comment by free-x** [comment]
@quirsh 
you can delete /etc/apt/sources.list.d/extra.list.save file
And please change /etc/apt/sources.list.d/extra.list
```
deb [signed-by=/usr/share/keyrings/oss.boating.gpg] https://www.free-x.de/ubuntu jammy main
deb [signed-by=/usr/share/keyrings/oss.boating.gpg] https://www.free-x.de/ubuntu jammy-preview main
```
Don't mix bullseye and jammy

**Comment by quirsh** [comment]
The change resulted in this error message from apt when I run `apt update`:
```
N: Skipping acquire of configured file 'main/binary-arm64/Packages' as repository 'https://www.free-x.de/ubuntu jammy InRelease' doesn't support architecture 'arm64'
N: Skipping acquire of configured file 'main/binary-arm64/Packages' as repository 'https://www.free-x.de/ubuntu jammy-preview InRelease' doesn't support architecture 'arm64'
```

**Comment by quirsh** [comment]
When I try to run avnav with the ochartsng plugin, the plugin always crashes and tries to restart. In the logs of avnav I find the following:
```
2024-05-26 16:13:30,652-2597-19426-Thread-3745 (handleProcessOutput)-INFO-PROVIDEROUT: working in /var/lib/avnav/ochartsng
2024-05-26 16:13:30,654-2597-19426-Thread-3745 (handleProcessOutput)-INFO-PROVIDEROUT: /usr/lib/avnav/plugins/ochartsng/AvnavOchartsProvider: error while loading shared libraries: libcrypto.so.1.1: cannot open shared object file: No such file or directory
```
Is this a problem because somewhere in the software ochartsng is still expecting the libssl to be version 1.1?

**Comment by wellenvogel** [comment]
Could you exactly describe how you installed?
And just check which openssl lib is there.
I tested on an ubuntu22 docker container and it did run without issues.

**Comment by quirsh** [comment]
Avnav was installed via apt, the ochartsng plugin was installed as described on the website for [daily build package installation](https://www.wellenvogel.net/software/avnav/docs/hints/ochartsng.html#h3:LinuxRaspberry):
```
wget -O avnav-ochartsng_20240525-raspbian-bullseye_arm64.deb https://www.wellenvogel.net/software/avnav/downloads/daily-ochartsng/20240525/avnav-ochartsng_20240525-raspbian-bullseye_arm64.deb
sudo apt install ./avnav-ochartsng_20240525-raspbian-bullseye_arm64.deb
sudo systemctl restart avnav
```
I just realized that I was forced to take the package from raspbian as there is no Ubuntu ARM64 package. Maybe that is the problem?

re openssl:
```user@host ~> openssl version
OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)```

**Comment by wellenvogel** [comment]
Ok - so you are running ubuntu on arm64.
Good to know...
Would not expect that it could work with some arm bullseye package...

**Comment by quirsh** [comment]
Yes, as stated in my initial post:

> My system is running Ubuntu 22.04 on a RaspberryPi4

When I looked for the daily builds I was so focused on the `*arm64.deb` ending that I didn't paid that much attention to the bullseye package. Is there a chance to get a arm64 Ubuntu package?

**Comment by wellenvogel** [comment]
I just saw that you mentioned the Pi...
Ok now the picture is more clear.
I guess we should do slightly better - currently we use the same dependency for building and for the final package.
For sure the raspbian-bullseye installs 1.1 for the build as it is still available.
But in the dependencies we still have the alternative of 1.1 or 3.
We should set the dependency to the version we used for the build.
So I guess we just would need a setup for ubuntu 22 arm64 - what we don't have right now.
Basically we could set up a build for that one - but at some point in time we could hit some cpu time limits on our builds.
Right now already >2h...
